### PR TITLE
Yaml options

### PR DESCRIPTION
**Pull request by fehguy**
Partial fix for #2292

### CODE REVIEW

**Comment by fehguy** [comment]
@webron this doesn't cover the quoted numeric keys but should get the others.  If good w/you, let's merge and keep #2292 open to track the numeric key issue (that existed in the previous major version too)

File: modules/swagger-core/src/main/java/io/swagger/util/ObjectMapperFactory.java [hunk 1]
```diff
@@ -24,7 +25,13 @@ protected static ObjectMapper createYaml() {
     }
 
     protected static ObjectMapper createYaml(boolean includePathDeserializer, boolean includeResponseDeserializer) {
-        return create(new YAMLFactory(), includePathDeserializer, includeResponseDeserializer);
+        YAMLFactory factory = new YAMLFactory();
+        factory.disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER);
+        factory.enable(YAMLGenerator.Feature.MINIMIZE_QUOTES);
+        factory.enable(YAMLGenerator.Feature.SPLIT_LINES);
+        factory.enable(YAMLGenerator.Feature.ALWAYS_QUOTE_NUMBERS_AS_STRINGS);
```
Comment (webron) [inline modules/swagger-core/src/main/java/io/swagger/util/ObjectMapperFactory.java]: Will this quote actual numeric values? Like `maxLength`?

Comment (fehguy) [inline modules/swagger-core/src/main/java/io/swagger/util/ObjectMapperFactory.java]: see the tests, only if they start that way

**Comment by fehguy** [comment]
Looks like the test failures are because we're expecting quoted strings in places where we're now getting unquoted ones.

**Pull request merged**